catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/ScriptFragment.kt

```diff
@@ -0,0 +1,1009 @@
+            .setTextWatcher(duplicateInputTextwatcher)
+            .setPositiveButton(
+                getString(R.string.ok)
+            ) { _: DialogInterface?, textInput: String? ->
+                pack(
+                    textInput,
+                    selectedBricks
+                )
+            }
+        builder.setTitle(R.string.new_group)
+            .setNegativeButton(R.string.cancel, null)
+            .show()
+    }
+
+    fun pack(name: String?, selectedBricks: List<Brick>?) {
+        try {
+            scriptController.pack(name, selectedBricks)
+            ToastUtil.showSuccess(activity, getString(R.string.packed_script_group))
+            switchToBackpack()
+        } catch (e: CloneNotSupportedException) {
+            Log.e(TAG, Log.getStackTraceString(e))
+        }
+        finishActionMode()
+    }
+
+    private fun switchToBackpack() {
+        val intent = Intent(activity, BackpackActivity::class.java)
+        intent.putExtra(BackpackActivity.EXTRA_FRAGMENT_POSITION, BackpackActivity.FRAGMENT_SCRIPTS)
+        startActivity(intent)
+    }
+
+    private fun switchToCatblocks() {
+        if (!BuildConfig.FEATURE_CATBLOCKS_ENABLED) {
+            return
+        }
+        var scriptIndex = -1
+        val firstVisible = listView?.firstVisiblePosition
+
+        if (listView?.count ?: 0 > 0 && firstVisible ?: -1 >= 0) {
+            val firstBrick = firstVisible?.let { listView?.getItemAtPosition(it) }
+            if (firstBrick is Brick) {
+                val scriptOfBrick = firstBrick.script
+                scriptIndex =
+                    projectManager.currentSprite.getScriptIndex(scriptOfBrick)
+            }
+        }
+
+        SettingsFragment.setUseCatBlocks(context, true)
+        val catblocksFragment = CatblocksScriptFragment(scriptIndex)
+        val fragmentTransaction = fragmentManager?.beginTransaction()
+        fragmentTransaction?.replace(
+            R.id.fragment_container, catblocksFragment,
+            CatblocksScriptFragment.TAG
+        )
+        fragmentTransaction?.commit()
+    }
+
+    private fun copy(selectedBricks: List<Brick>) {
+        val sprite = projectManager.currentSprite
+        brickController.copy(selectedBricks, sprite)
+        adapter.updateItems(sprite)
+        finishActionMode()
+    }
+
+    private fun showDeleteAlert(selectedBricks: List<Brick>) {
+        if (selectedBricks.isNotEmpty() && copyProjectForUndoOption()) {
+            showUndo(true)
+            undoBrickPosition = adapter.getPosition(selectedBricks[0])
+        }
+        delete(selectedBricks)
+    }
+
+    private fun delete(selectedItems: List<Brick>) {
+        val sprite = projectManager.currentSprite
+        brickController.delete(selectedItems, sprite)
+        adapter.updateItems(sprite)
+        finishActionMode()
+    }
+
+    private fun toggleComments(selectedBricks: List<Brick>) {
+        for (brick in adapter.getItems()) {
```

Comment:
```suggestion
        for (brick in adapter.items) {
```

Reply:
Same as above giving error while syncing with adapter PR.

Reply:
please share the Error.

Reply:
Here is the error after converting BrickAdapter.kt to kotlin in https://github.com/Catrobat/Catroid/pull/4246.

![image](https://user-images.githubusercontent.com/46495964/125192583-3a71a500-e266-11eb-9e60-5a263af40dc5.png)